lib/practice_exercises.rb

```diff
@@ -1,13 +1,61 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: is O(^mg)this_code_is_ridiculous an option? O(n)
+# Space Complexity: O(n^3)
 def is_palindrome(string)
```

Comment:
Your time complexity is O(3n) ==> O(n).  Since you're making several copies of the original string here but the memory usage only increases linearly as the array increases in size, so the space complexity is O(n).